client/src/containers/Login.js

```diff
@@ -48,6 +48,25 @@ class Login extends Component {
       );
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (!this.state.schema || !prevState.schema) {
```

Comment:
Initial thoughts are that this could use a few comments indicating the situations where either case here would fire